pages/CitationStyleLanguageHandler.inc.php

```diff
@@ -79,10 +79,19 @@ public function download($args, $request)
         $this->_setupRequest($args, $request);
 
         $plugin = PluginRegistry::getPlugin('generic', 'citationstylelanguageplugin');
-        $plugin->downloadCitation($request, $this->article, $this->citationStyle, $this->issue, $this->publication);
+        $plugin->downloadCitation($request, $this->submission, $this->citationStyle, $this->issue);
         exit;
     }
 
+    protected function isSubmissionUnpublished($issue, $submission) {
```

Comment:
Change the function signature to `($submission, $issue = null)`. That way if it is called outside of OJS, the caller doesn't need to pass the second argument.